test/plugin/scenarios/pulsar-scenario/src/main/java/test/apache/skywalking/apm/testcase/pulsar/controller/CaseController.java

```diff
@@ -51,38 +51,37 @@ public String pulsarCase() throws PulsarClientException, InterruptedException {
 
         String topic = "test";
 
+        CountDownLatch latch = new CountDownLatch(1);
+
         PulsarClient pulsarClient = PulsarClient.builder().serviceUrl(PULSAR_DOMAIN + serviceUrl).build();
 
         Producer<byte[]> producer = pulsarClient.newProducer().topic(topic).create();
 
-        Consumer<byte[]> consumer = pulsarClient.newConsumer().topic(topic).subscriptionName("test").subscribe();
+        Consumer<byte[]> consumer = pulsarClient.newConsumer().topic(topic)
```

Comment:
I noticed you are replacing the old one(based on the receive method) with the message listener one. We should cover all of these two cases in the test. Could you please add a new consumer or add a new test for this case. I think you can add a new consumer with a different subscription name which will get the same messages from the topic.

Reply:
Yes, because the `receive()` method will be called first before entering the `MessageListener` callback, so I directly replace the old one, and I think this can cover all cases in the test. I just add new expect data in `expectedData.yml` and reserve the `receive()` method segment. So based on this consideration, do I still need to create a new consumer to cover the new case?

Reply:
The consumer might disabled the message listener, it not a required feature for users to consume messages.

Reply:
OK, I will add a new consumer with message listener to test new case